packages/core/src/config/index.js

```diff
@@ -442,60 +455,72 @@ function parseHeadersEnvVar(envVarValue) {
 }
 
 /**
+ * Handles both stackTrace and stackTraceLength configuration
  * @param {InstanaConfig} config
  */
-function normalizeTracingStackTraceLength(config) {
-  if (config.tracing.stackTraceLength == null && process.env['INSTANA_STACK_TRACE_LENGTH']) {
-    parseStringStackTraceLength(config, process.env['INSTANA_STACK_TRACE_LENGTH']);
-  }
-  if (config.tracing.stackTraceLength != null) {
-    if (typeof config.tracing.stackTraceLength === 'number') {
-      config.tracing.stackTraceLength = normalizeNumericalStackTraceLength(config.tracing.stackTraceLength);
-    } else if (typeof config.tracing.stackTraceLength === 'string') {
-      parseStringStackTraceLength(config, config.tracing.stackTraceLength);
+function normalizeTracingStackTrace(config) {
+  const tracing = config.tracing;
+
+  const envStackTrace = process.env['INSTANA_STACK_TRACE'];
+  const envStackTraceLength = process.env['INSTANA_STACK_TRACE_LENGTH'];
+
+  // tracing.global?.stackTrace == stack trace mode (all, error, none)
+  if (tracing.global?.stackTrace !== undefined) {
+    const result = validateStackTraceMode(tracing.global.stackTrace);
+
+    if (result.isValid) {
+      tracing.stackTrace = configNormalizers.stackTrace.normalizeStackTraceMode(config);
+    } else {
+      logger.warn(`Invalid config.tracing.global.stackTrace: ${result.error}`);
+    }
+  } else if (envStackTrace !== undefined) {
+    const result = validateStackTraceMode(envStackTrace);
+
+    if (result.isValid) {
+      tracing.stackTrace = configNormalizers.stackTrace.normalizeStackTraceModeEnv(envStackTrace);
     } else {
+      logger.warn(`Invalid env INSTANA_STACK_TRACE: ${result.error}`);
+    }
+  }
+
+  // handles stackTraceLength
+  const isGlobalLengthDefined = tracing.global?.stackTraceLength !== undefined;
+  const isLegacyLengthDefined = tracing.stackTraceLength !== undefined;
+
+  if (isGlobalLengthDefined || isLegacyLengthDefined) {
+    if (isLegacyLengthDefined && !isGlobalLengthDefined) {
```

Comment:
If you have defined **isLegacyLengthDefined** -> we show the warning. It doesn't matter if customer already uses the new one.